### PR TITLE
implement windows gpu build

### PR DESCRIPTION
**Pull request by iameli**
**What does this pull request do? Explain your changes. (required)**

The recent change from `nvcc` to `clang` for the nvidia-flavored parts
of our ffmpeg build chain [1] basically gave us this feature for free.
All that was necessary was to add the mingw64 versions of the clang
compiler and it started happily producing binaries that interface with
CUDA on Windows.

I removed the Windows Docker build entirely. It was terrible, frequently
taking close to 90 minutes per build and frequently crashed. This change
implies that `docker/Dockerfile.build` and `docker/Dockerfile.build-linux`
can be combined back into one Dockerfile.build process; I'll open a tech
debt ticket for that but I don't think it's major enough to block this
merge.

Instead, we have a `.\windows-build.ps1` PowerShell script that takes
care of downloading MSYS2 using [Chocolatey](https://chocolatey.org/),
installing the necessary MSYS2/mingw64 packages, and running through
the build. It unpacks everything into a local .gitignored directory, so
it should be able to handle producing workspace-local binaries without
mucking with any system-level packages installed on the host.

This is the conclusion of a 10-month side project and I'm pretty stoked
😃

[1]: https://github.com/livepeer/go-livepeer/pull/1208

**How did you test each of these updates (required)**
I've tested this locally at my desk with a Windows T, Linux O, and macOS
B just to prove that it's possible. I have not done any kind of extensive
benchmarking work, but I did confirm that the `ja/lb` changes functioned
as expected with the appropriate Nvidia drivers. I think more testing is
definitely necessary, but shipping this PR and getting the binaries onto
more Windows computers is almost a prerequisite. Perhaps if we distribute
any Windows binaries we could label them as "experimental" or some such?

(I do think it's all working quite well, though.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,8 +30,15 @@ Alternative Livepeer installation options are also available:
 
 #### Building on Windows
 
-Building on Windows possible using Docker. Docker should run in 'Windows containers' mode.
-To build on Windows, just run `.\docker\windowsbuild.ps1`. This should create file `livepeer-windows-amd64.zip`, containig `livepeer.exe` and needed dlls.
+Buliding on Windows is possible using MSYS2 and mingw64. For convience, there is an `.\windows-build.ps1` script
```
Comment (mk-livepeer) [inline README.md]: Building

File: README.md [hunk 2]
```diff
@@ -30,8 +30,15 @@ Alternative Livepeer installation options are also available:
...
+that initializes an appropriate MSYS2 environment in the `.\msys2` directory. Steps to use:
+
+1. Install [Chocolatey](https://chocolatey.org/)
+2. Run `.\windows-build.ps1`
+3. Wait around 20 minutes
+
+If you run into problems, it's reccomended to delete your local `.\msys2` directory and re-run the script to rebuild
```
Comment (mk-livepeer) [inline README.md]: recommended

**Review by mk-livepeer — COMMENTED** [review]
I found a few typos.  Everything else looks reasonable.  I assume it all works.  Does this get built and tested in CI anywhere yet?

**Comment by iameli** [comment]
> Does this get built and tested in CI anywhere yet?

@mk-livepeer It gets built in Azure DevOps. It doesn't get tested very much, though it does do a basic `livepeer.exe -version` just to make sure the file can execute.

**Comment by iameli** [comment]
Typos fixed, rebased.

**Comment by iameli** [comment]
Rebased off master.

**Comment by iameli** [comment]
Rebased and integrated with the new build process. I'd like to get this merged soon — I've been using Windows binaries successfully on two computers for the last month or so.

**Comment by iameli** [comment]
@mk-livepeer @darkdarkdragon @j0sh  Could I get a :+1: on this from one of y'all? It works well and we have at least one community request for a Windows version.

**Review by darkdarkdragon — APPROVED** [review]
I've checked on Windows machine - it works!
LGTM!

**Comment by iameli** [comment]
Thanks @darkdarkdragon!

**Pull request merged**